src/lib.rs

```diff
@@ -257,14 +255,17 @@ impl<K: EnrKey> Enr<K> {
         self.seq
     }
 
+    /// # Panics
+    ///
+    /// Will panic if data is not sanitized
     /// Reads a custom key from the record if it exists, decoded as data.
     #[allow(clippy::missing_panics_doc)]
-    pub fn get(&self, key: impl AsRef<[u8]>) -> Option<&[u8]> {
+    pub fn get(&self, key: impl AsRef<[u8]>) -> Option<Vec<u8>> {
         // It's ok to decode any valid RLP value as data
-        self.get_raw_rlp(key).map(|rlp_data| {
-            rlp::Rlp::new(rlp_data)
-                .data()
+        self.get_raw_rlp(key).map(|mut rlp_data| {
+            Bytes::decode(&mut rlp_data)
                 .expect("All data is sanitized")
+                .to_vec()
```

Comment:
Public API of `get` does not need to change
```suggestion
                .as_ref()
```